domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/AmSwitchValidator.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.ccd.definition.store.domain;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.InvalidPropertyException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AmSwitchValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AmSwitchValidator.class);
+
+    public void validateAmPersistenceSwitchesIn(ApplicationParams appParams) {
+        List<String> duplicateForWriteDestinations = new ArrayList<>();
+        List<String> allForWriteDestinations = new ArrayList<>();
```

Comment:
Why do you need to keep track of `allForWriteDestinations` - keeping just `duplicateForWriteDestinations` is enough here I think?

Reply:
See https://github.com/hmcts/ccd-data-store-api/blob/RDM-4866/src/main/java/uk/gov/hmcts/ccd/data/caseaccess/AMSwitch.java and let me know.